lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -352,6 +358,32 @@ cardanoEra = cardanoEraFromRecentEra $ recentEra @era
 shelleyBasedEra :: forall era. IsRecentEra era => Cardano.ShelleyBasedEra era
 shelleyBasedEra = shelleyBasedEraFromRecentEra $ recentEra @era
 
+data MaybeInRecentEra (thing :: Type -> Type)
+    = InNonRecentEraByron
+    | InNonRecentEraShelley
+    | InNonRecentEraAllegra
+    | InNonRecentEraMary
+    | InNonRecentEraAlonzo
+    | InRecentEraBabbage (thing BabbageEra)
+    | InRecentEraConway (thing ConwayEra)
```

Comment:
What if we had something like this:
```hs
data MaybeInRecentEra
    = InBygoneEra BygoneEra
    | InRecentEra RecentEra

data BygoneEra
    = BygoneEraByron
    | BygoneEraShelley
    | BygoneEraAllegra
    | BygoneEraMary
    | BygoneEraAlonzo

data RecentEra
    = RecentEraBabbage
    | RecentEraConway
```

Perhaps it would make pattern matches on `MaybeInRecentEra` a little easier to write?

Reply:
It would make sense, but probably keeping it flat for now until we see a stronger need for it.